### PR TITLE
[ePD] Add total UNICEF cash amount without HQ contribution

### PR DESCRIPTION
**Pull request by roman-karpovich**
[ch24259]

### CODE REVIEW

**Review by reinbach — APPROVED** [review]

**Comment by roman-karpovich** [comment]
@reinbach please re-check. added total_hq_cash_local field to calculate HQ support amount

**Review by reinbach — APPROVED** [review]

File: src/etools/applications/partners/models.py [hunk 1]
```diff
@@ -2626,6 +2626,14 @@ class InterventionBudget(TimeStampedModel):
         verbose_name=_("Programme Effectiveness (%)"),
         default=0,
     )
+    total_hq_cash_local = models.DecimalField(
+        max_digits=20, decimal_places=2, default=0,
+        verbose_name=_('Total HQ Cash Local')
+    )
+    total_unicef_cash_local_wo_hq = models.DecimalField(
```
Comment (robertavram) [inline src/etools/applications/partners/models.py:2633]: this should be the current `self.total_unicef_contribution_local()` however that function must now change names to reflect that this contribution is wo HQ contrib

add another function `self.total_unicef_contribution_local()` that includes the total wo hq and add the hq amt(`total_hq_cash_local`) on top. There is no in-system direct relationship between the hq rate and the hq amt as the hq amt is now impossible to calculate automatically due to the rules.

File: src/etools/applications/partners/models.py [hunk 2]
```diff
@@ -2700,6 +2708,16 @@ def init_totals():
...
 
+        self.total_hq_cash_local = 0
+        self.total_unicef_cash_local_wo_hq = self.total_unicef_contribution_local()
+        if self.intervention.hq_support_cost:
+            self.total_unicef_cash_local_wo_hq /= decimal.Decimal(1 + self.intervention.hq_support_cost * 0.01)
+            self.total_hq_cash_local = self.total_unicef_cash_local_wo_hq
+            self.total_hq_cash_local *= decimal.Decimal(self.intervention.hq_support_cost * 0.01)
+            # get rid of extra decimal places according to field specs
+            self.total_hq_cash_local = self.total_hq_cash_local.quantize(decimal.Decimal('.01'))
+            self.total_unicef_cash_local_wo_hq = self.total_unicef_cash_local_wo_hq.quantize(decimal.Decimal('.01'))
```
Comment (robertavram) [inline src/etools/applications/partners/models.py:2719]: no need to quantize as the db field sets the two decimals

File: src/etools/applications/partners/models.py [hunk 3]
```diff
@@ -2700,6 +2708,16 @@ def init_totals():
         for item in self.intervention.supply_items.all():
             self.in_kind_amount_local += item.total_price
 
+        self.total_hq_cash_local = 0
+        self.total_unicef_cash_local_wo_hq = self.total_unicef_contribution_local()
+        if self.intervention.hq_support_cost:
```
Comment (robertavram) [inline src/etools/applications/partners/models.py:2713]: this logic block needs major refactor based on the changes

**Review by robertavram — CHANGES_REQUESTED** [review]
changes to the requirements

**Comment by acory** [comment]
@robertavram @domdinicola  
I tested the integration with FE (https://github.com/unicef/intervention-tab-pages/pull/341) and it works. Ready for merge.

**Pull request merged**